### DISCUSSION TITLE
FileUpload required validation

### DISCUSSION THREAD

**Discussion by NazarAli** [filamentphp/filament#15116] [General]
the required validation not work in FileUpload

> 
                        FileUpload::make('attachments')
                            ->label(__('global.attachments'))
                            ->directory('study-requests-attachments')
                            ->rules(['required'])
                            ->disk('public')
                            ->acceptedFileTypes(['application/pdf'])
                            ->downloadable()
                            ->openable()
                            ->panelLayout('grid')
>

**Comment by enessvg** [comment]
Please try this
```php
// ->rules(['required'])
->required()
```

**Reply by enessvg** [reply]
@NazarAli  If this helps, could you please mark it as the correct answer?

**Reply by NazarAli** [reply]
thank bro it work but why not work when using 

`->rules(['required'])`

**Reply by enessvg** [reply]
> thank bro it work but why not work when using
> 
> `->rules(['required'])`

Honestly, I have no idea. Minor problems with filamentphp

**Reply by enessvg** [reply]
@NazarAli hello if you wouldn't mind marking it as a solution?
